statistics/handle/handle.go

```diff
@@ -1007,12 +1007,21 @@ func (h *Handle) statsCacheLen() int {
 	return h.statsCache.Load().Len()
 }
 
-// updateStatsCache overrides the global statsCache with a new one, it may fail
+func (h *Handle) initStatsCache(newCache *cache.StatsCache) {
+	h.statsCache.Replace(newCache)
+	handle_metrics.CostGauge.Set(float64(h.statsCache.Load().Cost()))
+}
+
+// copyAndUpdateStatsCache overrides the global statsCache with a new one, it may fail
 // if the global statsCache has been modified by others already.
 // Callers should add retry loop if necessary.
-func (h *Handle) updateStatsCache(newCache *cache.StatsCache) (updated bool) {
-	h.statsCache.Replace(newCache)
-	handle_metrics.CostGauge.Set(float64(newCache.Cost()))
+func (h *Handle) copyAndUpdateStatsCache(newCache *cache.StatsCache, tables []*statistics.Table, deletedIDs []int64, opts ...cache.TableStatsOpt) (updated bool) {
```

Comment:
How about using `updateStatsCache` since this method doesn't always copy this cache...?

Reply:
Done